dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/scheduling/DefaultSchedulingManager.java

```diff
@@ -105,7 +107,20 @@ public void jobConfigurationStarted( JobConfiguration jobConfiguration )
     public void jobConfigurationFinished( JobConfiguration jobConfiguration )
     {
         runningJobConfigurations.remove( jobConfiguration );
-        jobConfigurationService.updateJobConfiguration( jobConfiguration );
+
+        JobConfiguration tempJobConfiguration = jobConfigurationService
+            .getJobConfigurationByUid( jobConfiguration.getUid() );
+
+        if ( tempJobConfiguration != null )
```

Comment:
Combine in one && statement

Reply:
It changes the semantics, so I dont think I can do that? 

This is the case where someone tried to stop a job(hence it will change in the db), but the job did not actually stop so it tries to update it when it finishes with an old object.

Reply:
You are right.